### PR TITLE
simplify track

### PR DESCRIPTION
**Pull request by gerhardol**
Implementing suggestions in #682 
Fixup for #836, with review comments (and a little more)

move preferences to Recording
Split "when" array to checkboxes
Create detail activity menu item Simplify
Enable for manual export GPS (not just FileSync)
Some touchups for  strings

@dratasich OK to squash into your changes and merge? Can you review?

### CODE REVIEW

File: app/src/main/org/runnerup/export/format/GPX.java [hunk 1]
```diff
@@ -208,7 +201,10 @@ private void exportLaps(long activityId, long startTime) throws IOException {
                                 }
                                 mXML.comment(" State change: " + locType + " " + formatTime(time));
                             }
-                        } else if (time > last_time) {
+                        } else if (time > last_time &&
+                                // ignore IDs that have been marked unnecessary by path simplification
+                                // reduces resolution of the exported path
+                                ! ignoreIDs.contains(cLocation.getInt(15))) {
```
[the remark this answers is not in the document]
Comment (dratasich) [reply app/src/main/org/runnerup/export/format/GPX.java]: There is still following bug:

original track | simplified track (enabled on export gpx)
:----:|:---:
Screenshot_20200106-162548 | Screenshot_20200106-160022

The total distance will be wrong in the gpx export.
I think I have to consider the lap starts and stops somehow.

[the remark this answers is not in the document]
Comment (gerhardol) [reply app/src/main/org/runnerup/export/format/GPX.java]: The "lap issue" is due to the use of segments. There is no direct concept of "laps" in gpx, RU and other fitness trackers often use segments to give some kind of notion of laps.
Similar problem for the summary: Other SW may consider time in between segments as pauses. The activity distance (really the time too, decreases precision) is recalculated for RU db changes (with my changes), the distance in GPX is not from the RU summary though. 

Two ways to handle this in GPX:
 * Avoid segments if there are gaps longer than GPS sampling time (several problems here, GPS can be temporary lost and the sampling time can be changed) or make configurable
 * Make sure points before/after lap start/stop is retained.

File: app/src/main/org/runnerup/export/format/GPX.java [hunk 2]
```diff
@@ -19,6 +19,7 @@
 
 import android.database.Cursor;
 import android.database.sqlite.SQLiteDatabase;
+import android.util.Log;
```
[the remark this answers is not in the document]
Comment (gerhardol) [reply app/src/main/org/runnerup/export/format/GPX.java]: Removing

File: app/src/main/org/runnerup/export/format/GPX.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (gerhardol) [reply app/src/main/org/runnerup/export/format/GPX.java]: Latest commit change so trkseg are not added if the track is being simplified.
Will not be optimal if a track has been simplified and the option is not set or for the uploaders using gpx.

Also adding an AlertDialog for the menu item to simplify.

With this, I believe this is ready to release as a new beta.

**Comment by gerhardol** [comment]
I pushed a squashed update, basically moving the recompute popup before the simplify and keeping the trkseg hack last. I would like to merge, to create a beta and get translations.

**Pull request merged**